### PR TITLE
Axum

### PR DESCRIPTION
**Pull request by cloudhead**

### CODE REVIEW

File: http-api/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ deadpool = "0.7.0"
 librad = { version = "0.1" }
 lnk-identities = { version = "0" }
 shared = { path = "../shared", default-features = false }
-warp = { version = "0.3.1", features = ["tls"] }
+#warp = { version = "0.3.1", features = ["tls"] }
```
Comment (cloudhead) [inline http-api/Cargo.toml:15]: Can remove

File: http-api/Cargo.toml [hunk 2]
```diff
@@ -31,6 +31,10 @@ async-trait = "0.1"
 ethers-core = "0.6.3"
 fastrand = "1.7.0"
 chrono = { version = "0.4.19", features = ["serde"] }
+axum = { version = "0.5.3", features = [ "headers" ] }
```
Comment (cloudhead) [inline http-api/Cargo.toml:34]:
```suggestion
axum = { version = "0.5.3", features = ["headers"] }
```

File: http-api/Cargo.toml [hunk 3]
```diff
@@ -31,6 +31,10 @@ async-trait = "0.1"
...
+axum-server = { version = "0.3", features = ["tls-rustls"] }
+hyper = { version ="0.14.17", features = ["server"] }
+tower-http = { version = "0.3.0", features = ["trace", "cors"] }
```
Comment (cloudhead) [inline http-api/Cargo.toml:37]: Did we check if these work with `default-features = false` as well? maybe saves a bit of build time.

**Review by cloudhead — COMMENTED** [review]
Looks good! Just a few very minor things.

**Comment by cloudhead** [comment]
Can you rebase and fix conflicts?

File: http-api/Cargo.toml (hunk 3 above)
Comment (xphoniex) [reply http-api/Cargo.toml:37]: Hm... I didn't, but it's pretty fast tbh.

**Pull request merged**